File: src/main/java/com/example/indie91/Models/InfluencerProfile.java
package com.example.indie91.Models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "influencer_profile")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class InfluencerProfile {

    @Id
    @GeneratedValue
    @Column(columnDefinition = "uuid", updatable = false, nullable = false)
    private UUID id;

    @Column(name = "total_earnings")
    private BigDecimal totalEarnings;

    @Column(name = "current_earnings")
    private BigDecimal currentEarnings;

    @Column(name = "last_paid_at")
    private LocalDateTime lastPaidAt;

    @Column(name = "total_sales")
    private Integer totalSales;

    @Column(name = "like_count")
    private Integer likeCount;

    @Column(name = "engagement_rate")
    private Double engagementRate;

    @Column(name = "brand_collaborations")
    private Integer brandCollaborations;

    @Column(name = "_content_count")
    private Integer underscoreContentCount;

    @Column(name = "content_count")
    private Integer contentCount;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    // Choose one of these based on actual usage:
    @Column(name = "user_id")
    private UUID userId;

    // If you actually use user_id (varchar), then:
    // @Column(name = "user_id")
    // private String userIdString;
}

File: models/influencer_profile.go
package models

import (
	"time"
)

type InfluencerProfile struct {
	ID                   string     `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`
	TotalEarnings        float64    `gorm:"type:decimal(38,2);default:0.00" json:"total_earnings"`
	CurrentEarnings      float64    `gorm:"type:decimal(38,2);default:0.00" json:"current_earnings"`
	LastPaidAt           *time.Time `gorm:"type:timestamp(6)" json:"last_paid_at"`
	TotalSales           int        `gorm:"default:0" json:"total_sales"`
	LikeCount            int        `gorm:"default:0" json:"like_count"`
	EngagementRate       float64    `gorm:"default:0.0" json:"engagement_rate"`
	BrandCollaborations  int        `gorm:"default:0" json:"brand_collaborations"`
	CreatedAt            *time.Time `gorm:"default:now()" json:"created_at"`
	UpdatedAt            *time.Time `gorm:"default:now()" json:"updated_at"`
	UserID               *string    `gorm:"type:uuid" json:"user_id"`
	ContentCount         int        `json:"content_count"`
	FollowersCount       int        `json:"followers_count"`

	AppUser AppUser `gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE;" json:"-"`
}

